node/subsystem/src/lib.rs

```diff
@@ -219,9 +219,33 @@ pub trait SubsystemContext: Send + 'static {
 ///
 /// [`Overseer`]: struct.Overseer.html
 /// [`Subsystem`]: trait.Subsystem.html
+#[async_trait]
 pub trait Subsystem<C: SubsystemContext> {
+	/// This subsystem's name
+	const NAME: &'static str;
+
+	/// Run this subsystem's main loop.
+	async fn run(self, ctx: C) -> SubsystemResult<()>;
+
 	/// Start this `Subsystem` and return `SpawnedSubsystem`.
-	fn start(self, ctx: C) -> SpawnedSubsystem;
+	///
+	/// This is implemented by default in terms of `run`.
```

Comment:
I think it would be useful to orient the client on which method they need to call under which circumstances and the implementer which method to implement under which circumstances.